test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ServiceBusSessionsEndToEndTests.cs

```diff
@@ -0,0 +1,571 @@
+
+                public override async Task CompleteProcessingMessageAsync(IMessageSession session, Message message, Executors.FunctionResult result, CancellationToken cancellationToken)
+                {
+                    _logger?.LogInformation("Custom processor End called!" + ServiceBusSessionsTestHelper.GetStringMessage(message));
+                    await base.CompleteProcessingMessageAsync(session, message, result, cancellationToken);
+                }
+            }
+
+            private Task ExceptionReceivedHandler(ExceptionReceivedEventArgs eventArgs)
+            {
+                return Task.CompletedTask;
+            }
+        }
+
+        public void Dispose()
+        {
+            if (_waitHandle1 != null)
+            {
+                _waitHandle1.Dispose();
+            }
+            if (_waitHandle2 != null)
+            {
+                _waitHandle2.Dispose();
+            }
+        }
+    }
+
+    internal class ServiceBusSessionsTestHelper
+    {
+        private static SessionHandlerOptions sessionHandlerOptions = new SessionHandlerOptions(ExceptionReceivedHandler);
+
+        public static async Task WriteQueueMessage(string connectionString, string queueName, string message, string sessionId = null)
+        {
+            QueueClient queueClient = new QueueClient(connectionString, queueName);
+            Message messageObj = new Message(Encoding.UTF8.GetBytes(message));
+            if (!string.IsNullOrEmpty(sessionId))
+            {
+                messageObj.SessionId = sessionId;
+            }
+            await queueClient.SendAsync(messageObj);
+            await queueClient.CloseAsync();
+        }
+
+        public static async Task WriteTopicnMessage(string connectionString, string topicName, string message, string sessionId = null)
+        {
+            TopicClient client = new TopicClient(connectionString, topicName);
+            Message messageObj = new Message(Encoding.UTF8.GetBytes(message));
+            if (!string.IsNullOrEmpty(sessionId))
+            {
+                messageObj.SessionId = sessionId;
+            }
+            await client.SendAsync(messageObj);
+            await client.CloseAsync();
+        }
+
+        public static async Task CleanUpQueue(string connectionString, string queueName)
+        {
+            QueueClient queueClient = new QueueClient(connectionString, queueName);
+            queueClient.RegisterSessionHandler(ProcessMessagesInSessionAsync, sessionHandlerOptions);
+            await Task.Delay(20000);
+            await queueClient.CloseAsync();
+        }
+
+        public static async Task CleanUpSubscription(string connectionString, string topicName, string subscriptionName)
+        {
+            SubscriptionClient subClient = new SubscriptionClient(connectionString, topicName, subscriptionName);
+            subClient.RegisterSessionHandler(ProcessMessagesInSessionAsync, sessionHandlerOptions);
+            await Task.Delay(20000);
```

Comment:
We don't want the tests to hang for 20 seconds. Can you remove this delay and the one above? What is this cleanup needed for?

Perhaps the tests can just create the required queues/subscriptions and delete them when finished. That way cleanup like this won't be needed?

Reply:
I refactored the code. We can't recreate the entity as SB SDK we are using does not support it.